### DISCUSSION TITLE
How to retrieve umlaut characters from pacs?

### DISCUSSION THREAD

**Discussion by minditer1** [dcm4che/dcm4che#1071] [Q&A]
I am trying to use dcm4che (version 5.22.6) to query patient data from a specific PACS, using dcm4che as a maven dependency in a Spring Boot Java application. Everything is working well, except patient names that have german characters (umlauts). Instead of retrieving name that contains ö, dcm4che returns � instead. 

I did set UTF-8 character encoding in my Spring Boot application and backend does read and write successfully umlauts: ````<project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>````

Tag specific character set is added on every request: 
````
public static void setCharacterSet(Attributes attributes) {
        int tag = Tag.SpecificCharacterSet;
        var vr = ElementDictionary.vrOf(tag, attributes.getPrivateCreator(tag));
        attributes.setString(tag, vr, "ISO_IR 100");
}
````
Is there anything else that I'm missing?

**Comment by gunterze** [comment]
You must not change the value of _Specific Character Set (0008,0005)_ in the data set retrieved by any DIMSE service from an archive or read from a DICOM file. Just access directly value of _Patient's Name (0010,0010)_ as String, which will decode the binary value of the attribute to a Java String, according the value of _Specific Character Set (0008,0005)_.

**Reply by minditer1** [reply]
Hi. Unfortunately this does not fix my issue, I've commented out the above method (setCharacterSet) and Attributes.getString(Tag.PatientName) still returns � instead of umlaut character

**Reply by gunterze** [reply]
What shows the log using dcm4che's [findscu](https://github.com/dcm4che/dcm4che/blob/master/dcm4che-tool/dcm4che-tool-findscu/README.md) utility with `-rPatientName`?

**Comment by minditer1** [comment]
https://github.com/dcm4che/dcm4che/issues/1075#issuecomment-997868991
@nroduit I have looked in the response and it seems that codec is in the response but not the corect value. I receive ISO 646 instead of the "ISO_IR 100" codec. Is there anything I can do?

**Reply by minditer1** [reply]
Actually I fixed this by manually setting the specific character set in the response attributes. I was looking at this like a workaround but it seems it is a good fix

**Reply by gunterze** [reply]
verify if the Specific Character Set is correctly set in the received object, investigating the stored file using some DICOM dumper (e.g. dcm4che's dcmdump utility)
